### PR TITLE
I18n: Fix path for Crowdin upload 

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**
When uploading the source for translations via `.github/workflows/i18n-crowdin-upload.yml` the file was created in the root directory in Crowdin although we have a file hierarchy.

**Why do we need this feature?**
This update to the mentioned GH action fixes the destination of the uploaded file.

**Who is this feature for?**
Everyone working on our i18n.

**Which issue(s) does this PR fix?**:
Relates to #81571

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: .github/workflows/i18n-crowdin-upload.yml [hunk 1]
```diff
@@ -19,11 +20,14 @@ jobs:
...
         with:
           upload_sources: true
+          upload_sources_args: '--dest=public/locales/en-US/grafana.json'
           upload_translations: false
           download_translations: false
           create_pull_request: false
           base_url: 'https://grafana.api.crowdin.com'
           config: 'crowdin.yml'
+          source: 'public/locales/en-US/grafana.json'
+          translation: 'public/locales/%locale%/%original_file_name%'
```
Comment (joshhunt) [inline .github/workflows/i18n-crowdin-upload.yml:30]: Hmm... It is annoying that this is duplicated from the crowdin.yaml file. I wonder maybe if we can get rid of the paths specified in the crowdin.yaml file then? Or, is it still needed for the download script? 

Very annoying!

Comment (L-M-K-B) [reply .github/workflows/i18n-crowdin-upload.yml:30]: 🤔 I'll test this on Monday and come back with feedback then.

Comment (L-M-K-B) [reply .github/workflows/i18n-crowdin-upload.yml:30]: Removing the whole files section in _crowdin.yml_ leads to an error in the download action:
Screenshot 2024-02-05 at 11 56 50

Only removing `source` and `translation` from the cobfig leads to another error in the download action:
Screenshot 2024-02-05 at 11 47 53

Furthermore I tested what happens if I remove `source` and `translation` from the config but add both to the download action like I did for the upload and it works. So I tend to implement it that way in order to be consistent with the GH actions (see newest change). 

What do you think?

Comment (joshhunt) [reply .github/workflows/i18n-crowdin-upload.yml:30]: Yup - this seems like the least worst option 😅

**Review by joshhunt — APPROVED** [review]

**Pull request merged**